lib/sms/sms.c

```diff
@@ -298,21 +430,37 @@ void sms_uninit(void)
 			return;
 		}
 		LOG_INF("SMS client unregistered");
+
+		/* Unregister from AT commands notifications. */
+		(void)at_notif_deregister_handler(NULL, sms_at_handler);
 	}
 
 	/* Cleanup resources. */
 	at_params_list_free(&resp_list);
 
 	if (cmt_rsp.alpha != NULL) {
 		k_free(cmt_rsp.alpha);
```

Comment:
Should all of the memory freeing be made into a separate function. You do it many places, and if you need to change the code, there will a lot of places to change.

Reply:
Yep, it's on my TODO list.

Reply:
This has been cleaned up